packages/block-editor/src/hooks/anchor.js

```diff
@@ -67,41 +68,56 @@ export const withInspectorControl = createHigherOrderComponent(
 			const hasAnchor = hasBlockSupport( props.name, 'anchor' );
 
 			if ( hasAnchor && props.isSelected ) {
+				const isWeb = Platform.OS === 'web';
+				const textControl = (
+					<TextControl
+						className="html-anchor-control"
+						label={ __( 'HTML anchor' ) }
+						help={
+							<>
+								{ __(
+									'Enter a word or two — without spaces — to make a unique web address just for this heading, called an “anchor.” Then, you’ll be able to link directly to this section of your page.'
+								) }
+
+								<ExternalLink
+									href={
+										'https://wordpress.org/support/article/page-jumps/'
+									}
+								>
+									{ __( 'Learn more about anchors' ) }
+								</ExternalLink>
+							</>
+						}
+						value={ props.attributes.anchor || '' }
+						valuePlaceholder={
+							! isWeb ? __( 'Add an anchor' ) : null
+						}
+						onChange={ ( nextValue ) => {
+							nextValue = nextValue.replace( ANCHOR_REGEX, '-' );
+							props.setAttributes( {
+								anchor: nextValue,
+							} );
+						} }
+						autoCapitalize="none"
+						autoComplete="off"
+					/>
+				);
+
 				return (
 					<>
 						<BlockEdit { ...props } />
-						<InspectorAdvancedControls>
-							<TextControl
-								className="html-anchor-control"
-								label={ __( 'HTML anchor' ) }
-								help={
-									<>
-										{ __(
-											'Enter a word or two — without spaces — to make a unique web address just for this heading, called an “anchor.” Then, you’ll be able to link directly to this section of your page.'
-										) }
-
-										<ExternalLink
-											href={
-												'https://wordpress.org/support/article/page-jumps/'
-											}
-										>
-											{ __( 'Learn more about anchors' ) }
-										</ExternalLink>
-									</>
-								}
-								value={ props.attributes.anchor || '' }
-								onChange={ ( nextValue ) => {
-									nextValue = nextValue.replace(
-										ANCHOR_REGEX,
-										'-'
-									);
-									props.setAttributes( {
-										anchor: nextValue,
-									} );
-								} }
-								autoComplete="off"
-							/>
-						</InspectorAdvancedControls>
+						{ isWeb && (
+							<InspectorAdvancedControls>
+								{ textControl }
+							</InspectorAdvancedControls>
+						) }
+						{ ! isWeb && props.name === 'core/heading' && (
+							<InspectorControls>
+								<PanelBody title={ __( 'Heading settings' ) }>
```

Comment:
'Heading settings' might not be very scalable for the future, as other features that get added to the block inspector for Heading can't be put inside this panel. There's no way to access it from the Block's edit function.

Reply:
Agreed, we'll need a more scalable heading in the future. [Thomas requested](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1816#issuecomment-756302170) that we use the "Heading settings" title for the time being.

Reply:
> Thomas requested

[My comment below](https://github.com/WordPress/gutenberg/pull/27935#discussion_r561787532) is actually covering the process here too, thanks!